drivers/pinctrl/pinctrl_mchp_xec.c

```diff
@@ -12,7 +12,7 @@
 #include <zephyr/drivers/pinctrl.h>
 
 /** @brief All GPIO register as arrays of registers */
-struct gpio_regs {
+struct gpio_regs_t {
```

Comment:
> so that it doesn't confuse with gpio_regs structure already defined in mec172x_gpio.h in soc

then why is it redefined here?

Reply:
the pinctrl driver was updated to work with both mec15xx and mec172x and in that process the common definitions were moved to the pinctrl source file. Here is the related [PR 47971](https://github.com/zephyrproject-rtos/zephyr/pull/47971)

Reply:
Does that mean we include unnecessary stuff?

Reply:
okay, I will work to move the common stuff to `soc/arm/microchip_mec/common/reg/ `folder

Reply:
moved the common `gpio_regs` structure to `soc/arm/microchip_mec/common/reg/ folder`
updated pinctrl driver accordingly